R/MsBackendDataFrame.R

```diff
@@ -248,14 +248,31 @@ setReplaceMethod("smoothed", "MsBackendMemory", function(object, value) {
 })
 
 #' @rdname hidden_aliases
-setMethod("spectraData", "MsBackendMemory", function(object, columns) {
-    ## TODO @jo IMPLEMENT:
-    ## allow to specify columns. Have then to check if these columns exist,
-    ## if not call the respective method.
+#'
+#' @importFrom methods as
+setMethod("spectraData", "MsBackendDataFrame", function(object, columns) {
```

Comment:
I anticipated to use `columns = TRUE` as default (see [here](https://github.com/lgatto/Spectra/blob/master/R/MsBackend.R#L495)), which returns all the columns. It is generally easier to have an explicit default rather than play with `missing` in the code. But my default clearly didn't anticipate the union of actual and mandatory columns.

Reply:
I was more thinking of allowing the user to specify which columns he wants to get returned. With database backends in mind.

Reply:
Sure, that was my understanding too: default returns all, setting `columns` to a set of valid names will only return those.

Reply:
But then I find it confusing to have the default `columns = TRUE`. If I was a user and see the definitions `columns = TRUE` I would assume that this parameter takes only a `logical(1)` as input, and not either `TRUE` or a `character` of column names ... but that's just a minor thing anyway.

Reply:
Here's a explanation is issues with `missing()`. Instead of `TRUE`, we could use `NULL` (as in the video), or `spectraVariables(object)`, to make it explicit that it's a character with spectra variables.

Reply:
👍 `columns = spectraVariables(object)`